### PR TITLE
Bug 1885403: Improve transitionCSVState error logs

### PR DESCRIPTION
**Pull request by awgreene**
Problem: OLM has received complaints that a while transitioning a CSV
from state to state, OLM suppresses a number of helpful error logs.
Currently, OLM surfaces errors in the reconciler that can be addressed
with additional syncs. Errors that cannot be addressed with additional
syncs should be logged.

Solution: Log errors that are not surfaced in OLM logs via the
reconciler.

### CODE REVIEW

**Review by hasbro17 — APPROVED** [review]
/lgtm

File: pkg/controller/operators/olm/operator.go [hunk 1]
```diff
@@ -1327,7 +1328,6 @@ func (a *Operator) transitionCSVState(in v1alpha1.ClusterServiceVersion) (out *v
 
 	modeSet, err := v1alpha1.NewInstallModeSet(out.Spec.InstallModes)
 	if err != nil {
-		syncError = err
 		logger.WithError(err).Warn("csv has invalid installmodes")
 		out.SetPhaseWithEventIfChanged(v1alpha1.CSVPhaseFailed, v1alpha1.CSVReasonInvalidInstallModes, syncError.Error(), now, a.recorder)
```
Comment (ankitathomas) [inline pkg/controller/operators/olm/operator.go:1332]: If syncError is being removed here, shouldn't this be changed to err?

Comment (awgreene) [reply pkg/controller/operators/olm/operator.go:1332]: Yes - great catch.

Comment (awgreene) [reply pkg/controller/operators/olm/operator.go]: Resolved in the latest version of the PR!

**Review by ankitathomas — APPROVED** [review]

**Pull request merged**